cart/interfaces/controller/cartcontroller.go

```diff
@@ -161,6 +167,11 @@ func (cc *CartViewController) UpdateQtyAndViewAction(ctx context.Context, r *web
 		}
 	}
 
+	err = cc.applicationCartService.CheckAndAdjustStockInCart(ctx, r.Session())
```

Comment:
Imho we should enable this via feature flag - and per default it should be deactivated